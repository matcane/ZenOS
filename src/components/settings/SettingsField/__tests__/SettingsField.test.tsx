import { render } from "@testing-library/react-native";

import SettingsFields from "../SettingsField";

const item = {
  name: "Theme",
  icon: "@assets/icons/icons8-paint-palette.webp",
  path: "settings/modals/Theme",
};

jest.mock("@react-native-firebase/auth", () => {
  return {
    onAuthStateChanged: jest.fn(),
    signInWithEmailAndPassword: jest.fn(),
    createUserWithEmailAndPassword: jest.fn(),
    signOut: jest.fn(),
  };
});

jest.mock("@react-native-firebase/firestore", () => {
  return {
    collection: jest.fn(() => ({
      doc: jest.fn(() => ({
        get: jest.fn(() => Promise.resolve({ exists: true, data: () => ({}) })),
        set: jest.fn(() => Promise.resolve()),
        update: jest.fn(() => Promise.resolve()),
        delete: jest.fn(() => Promise.resolve()),
      })),
    })),
  };
});

describe("SettingsField component", () => {
  it("renders correctly", () => {
    const tree = render(<SettingsFields setting={item} isFirst={true} isLast={false} />);
    expect(tree).toMatchSnapshot();
  });
});
