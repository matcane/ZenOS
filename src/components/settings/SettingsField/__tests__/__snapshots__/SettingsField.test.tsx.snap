// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SettingsField component renders correctly 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    [
      {
        "backgroundColor": "#e0e0e0",
      },
      {
        "alignItems": "center",
        "flexDirection": "row",
        "flexGrow": 1,
        "gap": 10,
        "height": 60,
        "justifyContent": "space-between",
        "paddingLeft": 10,
      },
      {
        "borderTopEndRadius": 10,
        "borderTopStartRadius": 10,
      },
      false,
    ]
  }
>
  <Image
    source={1}
    style={
      {
        "height": 48,
        "width": 48,
      }
    }
  />
  <Text
    style={
      [
        {
          "color": "#212121",
        },
        {
          "fontSize": 16,
        },
        {
          "flexGrow": 1,
        },
      ]
    }
  >
    Theme
  </Text>
  <Text />
</View>
`;
